install/install.sh

```diff
@@ -193,8 +193,28 @@ detect_platform() {
     cygwin_nt*) platform="pc-windows-msvc";;
     # mingw is Git-Bash
     mingw*) platform="pc-windows-msvc" ;;
-    # use the statically compiled musl bins on linux to avoid linking issues.
-    linux) platform="unknown-linux-musl" ;;
+    linux)
+      # default to the statically compiled musl bins on linux
+      # to avoid linking issues.
+      platform="unknown-linux-musl"
+      ldd_verstr="$(ldd --version | head -n1)"
+      ldd_glibc_prefix="ldd (GNU libc) "
+      if test "${ldd_verstr#*$ldd_glibc_prefix}" != "$ldd_verstr"; then
+        glibc_version=$(echo "${ldd_verstr}" | sed -e "s/^${ldd_glibc_prefix}//")
+        # on installations with a new enough glibc, use the dynamically compiled binary
+        #
+        # use this to see potential max versions
+        # strings $(which starship) | grep -i glibc
+        #
+        # then use https://gist.github.com/fasterthanlime/17e002a8f5e0f189861c with
+        # `MAX_VER` set to the version from earlier e.g. GLIBC_2.33 => 2.33
+        min_version="2.18"
+        if [ "${glibc_version}" > "${min_version}" ] \
```

Comment:
This is redirection unless escaped: `\>` :warning: 
Even then it is lexical comparison between strings rather than numerical one:
```sh
~ $ [ 3.22 \> 3.3 ] && echo 'true' || echo 'false'
false
```

Reply:
We can't do numerical comparison, sadly, since glibc version numbers might sometimes be of x.y.z form

I guess the cleanest way is to split based on dot and explicitly compare the parts as integers

Reply:
> I guess the cleanest way is to split based on dot and explicitly compare the parts as integers

Right, plus making sure that missing semver segments default to zero. :+1:

Reply:
fixed. Right now we can just compare major and minor, if we ever need to compare patch versions we can add that comparison then.